tikv-client/src/main/java/com/pingcap/tikv/TiSession.java

```diff
@@ -162,7 +162,7 @@ public ExecutorService getThreadPoolForTableScan() {
         if (tableScanThreadPool == null) {
           tableScanThreadPool =
               Executors.newFixedThreadPool(
-                  conf.getTableScanConcurrency(),
+                  conf.getTableScanConcurrency() / conf.getPartitionPerSplit(),
```

Comment:
could you give some explanations of what problem do you want to solve by this change, and how/why it works?

Reply:
Okay, I  will add more comments.